libsolidity/analysis/ConstantEvaluator.cpp

```diff
@@ -74,3 +74,25 @@ void ConstantEvaluator::endVisit(Literal const& _literal)
 	if (!_literal.annotation().type)
 		m_errorReporter.fatalTypeError(_literal.location(), "Invalid literal value.");
 }
+
+void ConstantEvaluator::endVisit(Identifier const& _identifier)
+{
+	VariableDeclaration const *variableDeclaration = dynamic_cast<VariableDeclaration const *>(_identifier.annotation().referencedDeclaration);
+	if (!variableDeclaration)
+		return;
+	if (!variableDeclaration->isConstant())
+	{
+		m_errorReporter.fatalTypeError(_identifier.location(), "Identifier must be declared constant.");
+	}
+	else
+	{
+		ASTPointer<Expression> value = variableDeclaration->value();
+		if (value)
```

Comment:
In what case doesn't it have a value and still counts as valid for array length?

Reply:
Oh, it won't be valid in that case. WIll throw an error in else clause.

Reply:
Done